src/tests/mod.rs

```diff
@@ -1901,3 +1903,46 @@ macro_rules! generate_ops_tests {
         }
     };
 }
+
+#[test]
+fn fraction_from_float() {
+    macro_rules! test_for_smaller_t {
+        ( $($t:ty),*) => {
+            $(
+                let f = GenericFraction::<$t>::from(2.0);
+                assert_eq!(format!("{}", f), "2");
+                let f = GenericFraction::<$t>::from(0.5);
+                assert_eq!(format!("{}", f), "1/2");
+                let f = GenericFraction::<$t>::from(15978.649);
+                assert_eq!(format!("{}", f), "NaN");
+            )*
+        };
+    };
+
+    macro_rules! test_for_larger_t {
+        ( $($t:ty),*) => {
+            $(
+                let f = GenericFraction::<$t>::from(2.0);
+                assert_eq!(format!("{}", f), "2");
+                let f = GenericFraction::<$t>::from(0.5);
+                assert_eq!(format!("{}", f), "1/2");
+                let f = GenericFraction::<$t>::from(15978.649);
+                assert_eq!(format!("{}", f), "15978649/1000");
+            )*
+        };
+    };
+    test_for_smaller_t!(u8, i8, u16, i16);
+    test_for_larger_t!(u32, i32, u64, i64, u128, i128, usize, isize);
```

Comment:
Rather than having it in a separate test below, I'd expect here to be another macro for limitless (heap allocated types). E.g. something like that:

```rust
#[cfg(feature = "with-bigint")]
{
    use super::{BigInt, BigUint};
    test_for_big_t!(BigUint, BigInt);
}
```